### PR TITLE
feat(base): creates project base and server

### PR DESCRIPTION
**Pull request by pallavi**
**What**: Creates project base and server for sfmovies

**Why**: So we can build on this to add endpoints

**Details**:
- Initializes sfmovies as a Node package
- Sets up config, Travis integration, linting rules, node version configuration, testing modules, and scripts for testing and release with the yeoman lob generator
- Creates a Hapi server and adds a script to run the server using nodemon

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: node_js
+sudo: false
+node_js:
+  - 6
+env:
+  - NODE_ENV="test"
+script:
+  - npm test
+  #- npm run enforce
```
Comment (pallavi) [inline .travis.yml]: temporarily commented out so Travis build passes

File: package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "chai": "^3.0.0",
+    "eslint": "^1.9.0",
+    "eslint-config-lob": "^2.0.0",
+    "generate-changelog": "^1.0.0",
+    "istanbul": "^0.4.2",
+    "mocha": "^2.2.5",
+    "nodemon": "^1.11.0"
+  },
+  "scripts": {
+    "lint": "eslint .",
```
Comment (sjl2) [inline package.json]: nit: alphabetize

File: .node-version [hunk 3]
```diff
@@ -0,0 +1 @@
+6
```
Comment (sjl2) [inline .node-version]: Won't matter here, but keep in mind that in most repos, we're going to set this to the most specific version we can to ensure it doesn't upgrade on us.

**Review by sjl2 — APPROVED** [review]
LGTM!

File: .node-version (hunk 3 above)
Comment (robinjoseph08) [reply .node-version]: for posterity, lets update it to `6.11.1` here and in the `.travis.yml`. its nice to be exact

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  - npm run lint
```
Comment (robinjoseph08) [inline .travis.yml]: can you change these `npm run`s to `yarn`? so itd be

```
- yarn test
# - yarn enforce
- yarn lint
```

yarn doesnt need the `run` cause its smarter than your average bear

File: package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "release:patch": "changelog -p && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version patch && git push origin && git push origin --tags",
+    "release:minor": "changelog -m && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version minor && git push origin && git push origin --tags",
+    "release:major": "changelog -M && git add CHANGELOG.md && git commit -m 'updated CHANGELOG.md' && npm version major && git push origin && git push origin --tags",
+    "start": "nodemon --ignore test/",
+    "test": "NODE_ENV=test istanbul cover _mocha -- test --require test/setup.js --recursive --timeout 30000",
+    "test:raw": "NODE_ENV=test mocha test --require test/setup.js --recursive --timeout 30000"
+  },
+  "optionalDependencies": {
+    "fsevents": "^1.1.2"
```
Comment (robinjoseph08) [inline package.json]: you should be able to just run `yarn remove fsevents` and not include it anywhere in the `package.json` and things should work

File: package.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  "dependencies": {
```
Comment (robinjoseph08) [inline package.json]: super nit: move `dependencies` before `devDependencies` since thats usually where we find them

**Review by robinjoseph08 — APPROVED** [review]
just some small stuff, but LGTM!

**Pull request merged**